### PR TITLE
added identity transformer

### PR DESCRIPTION
**Pull request by koaning**
Implementing this one was interesting ... 

```python
import pytest
from sklearn.utils import estimator_checks
from sklearn.preprocessing import FunctionTransformer

from sklego.common import flatten
from sklego.preprocessing import IdentityTransformer
from tests.conftest import transformer_checks, general_checks


@pytest.mark.parametrize(
    "test_fn",
    flatten(
        [
            transformer_checks,
            general_checks,
            # nonmeta_checks,
            estimator_checks.check_estimators_dtypes,
            estimator_checks.check_fit_score_takes_y,
            estimator_checks.check_dtype_object,
            estimator_checks.check_sample_weights_pandas_series,
            estimator_checks.check_sample_weights_list,
            estimator_checks.check_sample_weights_invariance,
            estimator_checks.check_estimators_fit_returns_self,
            estimator_checks.check_complex_data,
            estimator_checks.check_estimators_empty_data_messages,
            estimator_checks.check_pipeline_consistency,
            estimator_checks.check_estimators_nan_inf,
            estimator_checks.check_estimators_overwrite_params,
            estimator_checks.check_estimator_sparse_data,
            estimator_checks.check_estimators_pickle,
        ]
    ),
)
def test_estimator_checks(test_fn):
    test_fn(IdentityTransformer.__name__, IdentityTransformer())
    test_fn(FunctionTransformer.__name__, FunctionTransformer(func=identity, inverse=identity))


def identity(X):
    return X
```

It turns out that the `FunctionTransformer` does **not** pass all tests. 

Mine does.

### CODE REVIEW

File: sklego/preprocessing/identitytransformer.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.preprocessing import FunctionTransformer
+
+from sklearn.utils import check_array
+from sklearn.utils.validation import FLOAT_DTYPES, check_is_fitted
+
+
+def _identity(X):
```
Comment (MBrouns) [inline sklego/preprocessing/identitytransformer.py:8]: This can go right?

File: tests/test_preprocessing/test_identitytransformer.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        ]
+    ),
+)
+def test_estimator_checks(test_fn):
+    test_fn(IdentityTransformer.__name__, IdentityTransformer())
+    test_fn(FunctionTransformer.__name__, FunctionTransformer(func=identity))
+
+
+def identity(X):
+    return X
```
Comment (MBrouns) [inline tests/test_preprocessing/test_identitytransformer.py:40]: Doesn't `flake8` complain about missing newlines at the end of a file?

File: sklego/preprocessing/identitytransformer.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+class IdentityTransformer(BaseEstimator, TransformerMixin):
+    """
+    The identity transformer returns what it is fed. Does not apply anything useful.
+    The reason for having it is because you can build more expressive pipelines.
+    """
+
+    def fit(self, X, y=None):
+        """'Fits' the estimator."""
+        X = check_array(X, copy=True, estimator=self)
+        self.fitted_ = True
```
Comment (MBrouns) [inline sklego/preprocessing/identitytransformer.py:21]: `fitted_` isn't needed if you also check `shape_` right?

**Comment by koaning** [comment]
Crud. I thought I had WIP on. mybad.

File: sklego/preprocessing/identitytransformer.py (hunk 3 above)
Comment (koaning) [reply sklego/preprocessing/identitytransformer.py:20]: fixing now

**Comment by koaning** [comment]
@MBrouns should be all fixed now. Approved?

**Comment by MBrouns** [comment]
LGTM!

**Pull request merged**